app/js/components/Zhishi.react.js

```diff
@@ -18,9 +18,12 @@ class Zhishi extends React.Component {
   render(){
     return (
       <div className="full-height">
-        {this.props.children && React.cloneElement(this.props.children, {
-          app_state: this.state
-        })}
+        <SidebarPusher />
+        <div className="pusher">
+          {this.props.children && React.cloneElement(this.props.children, {
```

Comment:
I was asking about the use case of React.cloneElement here? it seems we could have just used the React component instead of cloneelement. Clone element is rarely used. and to some extent its considered a bad practice if one has to use it